siglens/siglens
# Fix for SQL Query Index

# Description
- This change ensures that the correct index is selected when running an SQL query. Previously, the code did not correctly handle the case where the index name was specified in the SQL query.

- The updated code checks if there exists an index name in the query. If true, it sets indexNameIn to aggs.TableName, which is where we are parsing the tableName and storing it in aggs.

- This change gives precedence to the index name specified in the SQL query over the index name selected from the dropdown in the UI. If there is no index name in the SQL query, the index name selected from the UI will be used. However, if an index name is specified in the SQL query, the index name selected from the UI will be ignored.

Fixes #197 

# Testing
- Run the SQL search. For example, if you've ingested data into an index named ind-2, you can run:
 `` SELECT * FROM `ind-2` ``
 Verify that only data from ind-2 is returned. If ind-2 does not exist, no data should be returned.
- Repeat for other index names to ensure that the correct index is always selected.
- Test the precedence of the index name by selecting a different index from the UI dropdown and running an SQL query with a specific index name. The data returned should be from the index specified in the SQL query, not the index selected from the dropdown.

Below image shows data being selected from `proddata-test` ![image](https://github.com/siglens/siglens/assets/113487319/9a2cc40f-6a71-4ecb-8e17-f85a11bf8cbd)

# Checklist:
Before marking your pull request as ready for review, complete the following.
